src/JustSaying/Messaging/MessageSerialization/TypedSystemTextJsonSerializationFactory.cs

```diff
@@ -0,0 +1,11 @@
+#if NET8_0_OR_GREATER
+using System.Text.Json;
+using JustSaying.Models;
+
+namespace JustSaying.Messaging.MessageSerialization;
+
+public class TypedSystemTextJsonSerializationFactory(JsonSerializerOptions options) : IMessageSerializationFactory
```

Comment:
Not the biggest fan of the `Typed` prefix.

Reply:
Yeah, same, I was at the "get it working at any cost" stage by then. I'll try and think of something a bit sleaker, _or_ merge this back in with the current STJSerializationFactory, and have it use the typed serializer when dynamic code isn't supported.

If we go this way, in the next breaking version, I think it'll make sense to deprecate the non-typed serializer, and to that end, maybe decorate it now.